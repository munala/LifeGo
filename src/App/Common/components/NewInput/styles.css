.input-container {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  align-self: stretch;
  margin-left: 5px;
  margin-right: 5px;
  margin-bottom: 10px;
  padding-left: 10px;
  padding-right: 10px;
}
.textField {
  display:flex;
  color:'#555';
  flex-grow:1;
}
