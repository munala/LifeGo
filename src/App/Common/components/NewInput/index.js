import React from 'react';
import PropTypes from 'prop-types';

import Avatar from '../Avatar';
import TextInput from '../TextInput';
import OutlinedButton from '../OutlinedButton';
import './styles.css';

const NewInput = ({
  pictureUrl,
  content,
  focus,
  onChange,
  save,
  editMode,
  submitting,
  cancel,
  buttonLabel,
}) => (
  <div className="input-container">
    <Avatar
      style={{ display: 'flex', height: 30, width: 30 }}
      src={
        pictureUrl || require('../../../../assets/images/user.png') // eslint-disable-line global-require
      }
    />
    <TextInput
      type="text"
      placeholder={`type ${content.type}`}
      disableUnderline
      onFocus={focus}
      value={content.content || ''}
      onChange={onChange}
      className="textField"
      style={{ display: 'flex' }}
      onKeyPress={(e) => {
        if (e.key === 'Enter') { save(); }
      }
}
    />
    <OutlinedButton
      onClick={save}
      disabled={submitting}
      label={buttonLabel}
      style={{ display: 'flex' }}
    />
    {
      editMode &&
      <div onClick={cancel}>
        Cancel
      </div>
    }
  </div>
);

NewInput.propTypes = {
  pictureUrl: PropTypes.string,
  content: PropTypes.shape({
    type: PropTypes.string.isRequired,
  }).isRequired,
  focus: PropTypes.func,
  onChange: PropTypes.func.isRequired,
  save: PropTypes.func.isRequired,
  editMode: PropTypes.bool,
  submitting: PropTypes.bool.isRequired,
  cancel: PropTypes.func,
  buttonLabel: PropTypes.string.isRequired,
};

NewInput.defaultProps = {
  pictureUrl: null,
  editMode: false,
  focus: () => {},
  cancel: () => {},
};

export default NewInput;
