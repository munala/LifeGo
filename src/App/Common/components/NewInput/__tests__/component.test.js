import React from 'react';
import { shallow } from 'enzyme';
import NewInput from '../';

describe('NewInput tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<NewInput
      buttonLabel="name"
      name="name"
      submitting={false}
      value="value"
      content={{ type: 'edit' }}
      save={() => {}}
      onChange={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
