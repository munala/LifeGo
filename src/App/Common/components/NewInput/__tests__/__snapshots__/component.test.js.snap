// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewInput tests renders without crashing 1`] = `
<div
  className="input-container"
>
  <UserAvatar
    name="user avatar"
    onClick={[Function]}
    src="user.png"
    style={
      Object {
        "display": "flex",
        "height": 30,
        "width": 30,
      }
    }
  />
  <WithStyles(TextInput)
    className="text-field"
    disableUnderline={true}
    name=""
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    placeholder="type edit"
    style={
      Object {
        "fontSize": 14,
      }
    }
    type="text"
    value=""
  />
  <a
    className="submit-button"
    onClick={[Function]}
  >
    NAME
  </a>
</div>
`;
